src/simpleNbox.cpp

```diff
@@ -145,14 +155,20 @@ void SimpleNbox::setData( const std::string &varName,
         else if( varNameParsed == D_VEGC ) {
             H_ASSERT( data.date == Core::undefinedIndex(), "date not allowed" );
             veg_c[ biome ] = data.getUnitval( U_PGC );
+            // HACK: Set value at t=0
```

Comment:
I think we can do better.

Reply:
I agree.

Does it make sense to use `t = data.date` if date is provided and `t = 0` if `date` is missing? Or do we just force users to provide the first year of the simulation?

Reply:
My gut reaction is they have to provide the date.

Reply:
I agree with @bpbond on this.  If something is time variable, then you should have to supply a date to get or set it.  The only places where we allow a default date for something that varies over time are places where it was done to maintain backward compatibility (specifically, variables where we used to keep only a current value, and model components relied on that behavior when exchanging data internally).

However, my remark referred to the comment itself.  Please don't put comments like "this is a hack" in your code; it's completely unhelpful to anyone else who comes along later and reads it.  It only takes a few minutes to make a note of what you were trying to do and why it had to be done in a hacky way.  Take those few minutes.